chipflow_lib/steps/silicon.py

```diff
@@ -113,158 +122,249 @@ def submit(self, rtlil_path, *, dry_run=False, wait=False):
+                        match line_str[0:8]:
+                            case "DEBUG   ":
+                                sp.info(line_str) if log_level <= logging.DEBUG else None
+                            case "INFO    ":
+                                sp.info(line_str) if log_level <= logging.INFO else None
+                                # Some special handling for more user feedback
+                                if line_str.endswith("started"):
+                                    steps = re.findall(r"([0-9a-z_.]+)\:+?", line_str[18:])[0:2]
+                                    sp.text = "Build running... " + ' -> '.join(steps)
+                            case "WARNING ":
+                                sp.info(line_str) if log_level <= logging.WARNING else None
+                            case "ERROR   ":
+                                sp.info(line_str) if log_level <= logging.ERROR else None
+                        sp.start()
+                else:
+                    stream_event_counter +=1
+                    logger.debug(f"Failed to stream logs: {log_resp.text}")
+                    sp.text = "💥 Failed streaming build logs. Trying again!"
+                    break
+            except requests.ConnectTimeout:
+                sp.text = "💥 Failed connecting to ChipFlow Cloud."
+                logger.debug(f"Error while streaming logs: {e}")
+                break
+            except requests.RequestException as e:
+                sp.text = "💥 Failed streaming build logs. Trying again!"
+                logger.debug(f"Error while streaming logs: {e}")
+                stream_event_counter +=1
+                continue
+            except requests.ConnectionError as e:
+                if type(e.__context__) is urllib3.exceptions.ReadTimeoutError:
+                    continue  #just timed out, continue long poll
+
+        # save steps so we coninue where we left off if we manage to reconnect
+        self._last_log_steps = steps
+        return stream_event_counter
+
+    def _stream_logs(self, sp, network_err):
+        sp.start("Streaming the logs...")
+        # Poll the status API until the build is completed or failed
+        fail_counter = 0
+        timeout = 10.0
+        build_status = "pending"
+        stream_event_counter = 0
+        self._last_log_steps = []
+        while fail_counter < 10 and stream_event_counter < 10:
+            sp.text = f"Waiting for build to run... {build_status}"
+            time.sleep(timeout)  # Wait before polling
+            try:
+                status_resp = requests.get(
+                    self._build_status_url,
+                    auth=(None, self._chipflow_api_key),
+                    timeout=timeout
+                )
+            except requests.ConnectTimeout as e:
+               network_err(e)
+            except requests.ConnectionError as e:
+                if type(e.__context__) is urllib3.exceptions.MaxRetryError:
```

Comment:
should this be handling the case where a `ConnectionError` is thrown but `e.__context__` isn't `MaxRetryError`, too?

Reply:
good catch, a bit of left over logic from what happens in _long_poll_stream..

Reply:
done!